iib/workers/tasks/build_merge_index_image.py

```diff
@@ -13,20 +13,89 @@
     _get_present_bundles,
     _get_resolved_bundles,
     _opm_index_add,
-    _prepare_request_for_build,
     _push_image,
     _update_index_image_build_state,
     _update_index_image_pull_spec,
 )
 from iib.workers.tasks.celery import app
-from iib.workers.tasks.utils import request_logger, run_cmd, set_registry_token
+from iib.workers.tasks.utils import (
+    request_logger,
+    run_cmd,
+    set_registry_token,
+    RequestConfigMerge,
+    get_all_index_image_infos,
+    gather_index_image_arches,
+    _validate_distribution_scope,
+    _get_resolved_image,
+    _get_image_arches,
+)
 
 
 __all__ = ['handle_merge_request']
 
 log = logging.getLogger(__name__)
 
 
+def _prepare_request_for_build(request_id, build_request_config):
+    """
+    Prepare the request for the index image build.
+
+    All information that was retrieved and/or calculated for the next steps in the build are
+    returned as a dictionary.
+
+    This function was created so that code didn't need to be duplicated for the ``add`` and ``rm``
+    request types.
+
+    :param int request_id: the ID of the IIB build request
+    :param RequestConfg build_request_config: build request configuration
+    :rtype: dict
+    :return: a dictionary with the keys: arches, binary_image_resolved, from_index_resolved, and
+        ocp_version.
+
+    :raises IIBError: if the container image resolution fails or the architectures couldn't be
+        detected.
+    """
+    set_request_state(request_id, 'in_progress', 'Resolving the container images')
+    index_image_infos = get_all_index_image_infos(
```

Comment:
Could you please rename `index_image_infos ` to `index_image_info`

Reply:
method is named infos. Wouldn't that be strange to have info?